cmd/jaeger/internal/integration/kafka_test.go

```diff
@@ -4,11 +4,63 @@
 package integration
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
 	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
 
 	"github.com/jaegertracing/jaeger/plugin/storage/integration"
 )
 
+func createConfigWithEncoding(t *testing.T, configFile string, targetEncoding string, uniqueTopic string) string {
+	data, err := os.ReadFile(configFile)
+	require.NoError(t, err, "Failed to read config file: %s", configFile)
+
+	var config map[string]any
+	err = yaml.Unmarshal(data, &config)
+	require.NoError(t, err, "Failed to unmarshal YAML data from config file: %s", configFile)
+
+	// Function to recursively search and replace the encoding and topic
+	var replaceEncodingAndTopic func(m map[string]any) int
+	replaceEncodingAndTopic = func(m map[string]any) int {
```

Comment:
```suggestion
	replaceEncodingAndTopic := func(m map[string]any) int {
```

Reply:
The replaceEncoding function is being referenced within itself, but when we declare the function using the shorthand approach (:=), it hasn't been fully defined at that point so wouldn't that be throwing up an error saying it's undefined?

Reply:
+1